mapping_parenting_tech/analysis/prototyping/clustering_descriptions.py

```diff
@@ -68,34 +76,371 @@
+    "uk.co.bbc.cbeebiesgoexplore",
+    "tv.alphablocks.numberblocksworld",
+    "com.acamar.bingwatchplaylearn",
+    "uk.org.bestbeginnings.babybuddymobile",
+    "com.mumsnet.talk",
+    "com.mushuk.mushapp",
+    "com.teampeanut.peanut",
+    "com.flipsidegroup.nightfeed.paid",
+]
+
+extra_apps = [
+    "org.twisevictory.apps",
+    "com.domustechnica.tww.sleep",
+    "com.domustechnica.backtoyou",
+    "twwaudioapp.qsd.com.twwaudio.v2",
+    "com.tinybeans",
+    "com.edokicademy.montessoriacademy",
+    "uk.co.bbc.cbeebiesplaytimeisland",
+    "uk.co.bbc.cbeebiesgoexplore",
+    "uk.co.bbc.cbeebiesgetcreative",
+    "air.uk.co.bbc.cbeebiesstorytime",
+    "com.teachyourmonstertoread.tmapp",
+    "abc_kids.alphabet.com",
+    "com.duckduckmoosedesign.kindread",
+    "com.teampeanut.peanut",
+    "com.ovuline.pregnancy",
+    "com.ovuline.parenting",
+    "com.pgs.emmasdiary",
+    "com.backthen.android",
+    "uk.co.happity.happity",
+    "uk.co.hoop.android",
+    "com.nighp.babytracker_android",
+    "com.amila.parenting",
+    "uk.org.bestbeginnings.babybuddymobile",
+    "com.rvappstudios.baby.games.piano.phone.kids",
+    "com.educational.baby.games",
+    "com.happytools.learning.kids.games",
+    "org.msq.babygames",
+    "com.thepositivebirthcompany.freyasurgetimer",
+    "com.mathletics",
+    "com.huckleberry_labs.app",
+    "com.propagator.squeezy",
+    "com.sitekit.eRedBook",
+]
+
+# %%
+cluster_options = {
+    "relevant": {
+        2: "period and fertility tracking",
+        7: "maths games + science education",
+        8: "baby sleep",
+        9: "learning [foreign?] language",
+        10: "drawing / colouring in",
+        12: "parent support / keeping in touch / photo sharing",
+        13: "baby 'habit' tracking, inc. breastfeeding",
+        17: "stories",
+        18: "learning to read and write",
+        19: "ABCs",
+        27: "baby/toddler educational games",
+    },
+    "interesting": {
+        5: "music apps, inc for toddlers",
+        11: "young games",
+        28: "young games",
+    },
+    "unlikely": {
+        6: "student learning",
+        16: "being organised/time management",
+        20: "medical/health support (for adults)",
+        22: "meditation/mindfulness/rest",
+        25: "positivity/self development",
+        26: "TBC",
+    },
+    "discard": {
+        0: "Russian",
+        1: "religion",
+        3: "driving test",
+        4: "web safety",
+        14: "outdoor / mapping",
+        15: "engineering/physics",
+        21: "medical education",
+        23: "medical 'organising'",
+        24: "medical",
+    },
+}
+
+# %%
+relevant_clusters = set()
+for a_id in set(extra_apps + apps_of_interest):
+    if a_id in df.appId.values:
+        relevant_clusters.update(df[df.appId == a_id].cluster.values)
+        if df[df.appId == a_id].cluster.values == -1:
+            print(f"{a_id} not in cluster")
+
+relevant_clusters
+
+
+# %%
+def get_cluster_info(needle: int, haystack: dict):
```

Comment:
Hmm, probably should use a bit more informative variable names ;)

Reply:
Have genuinely seen these variable names used in built-in functions - "looking for a needle in a haystack" is a common figure of speech? Will add details to the function's docstring